### PR TITLE
3529 fix date range cover last day

### PR DESCRIPTION
**Pull request by groony**

### CODE REVIEW

**Comment by groony** [comment]
#3529

**Comment by timoschilling** [comment]
please rename `lteqdate` to `lteq_date` and provide the same for `gteq`

File: lib/ransack_ext.rb [hunk 1]
```diff
@@ -9,4 +9,8 @@
   {'equals'=>'eq', 'greater_than'=>'gt', 'less_than'=>'lt'}.each do |old,current|
     config.add_predicate old, arel_predicate: current
   end
+
+  config.add_predicate 'lteqdate',
+  	arel_predicate: 'lt',
+  	formatter: -> (v) { v + 1.day }
```
Comment (timoschilling) [inline lib/ransack_ext.rb]: Way not `v + 1.day - 1.second`, like in the previews commit?

**Comment by groony** [comment]
@timoschilling done

File: lib/ransack_ext.rb (hunk 1 above)
Comment (groony) [inline lib/ransack_ext.rb]: For  ``v + 1.day - 1.second`` predicate **lteq**, but for ``v + 1.day`` predicate **lt**
I think second variant better,

File: lib/ransack_ext.rb [hunk 2]
```diff
@@ -9,4 +9,11 @@
   {'equals'=>'eq', 'greater_than'=>'gt', 'less_than'=>'lt'}.each do |old,current|
     config.add_predicate old, arel_predicate: current
   end
+
+  config.add_predicate 'gteq_date',
+    arel_predicate: 'gteq'
+
+  config.add_predicate 'lteq_date',
+  	arel_predicate: 'lt',
+  	formatter: -> (v) { v + 1.day }
```
Comment (timoschilling) [inline lib/ransack_ext.rb]: Way not v + 1.day - 1.second, like in the previews commit?

Comment (groony) [inline lib/ransack_ext.rb]: For  ``v + 1.day - 1.second`` predicate **lteq**, but for ``v + 1.day`` predicate **lt**
I think second variant better,

Comment (timoschilling) [inline lib/ransack_ext.rb]: ok

**Comment by timoschilling** [comment]
I will think about this later this day or at the weekend

**Comment by groony** [comment]
ok, tnx

**Comment by seanlinsley** [comment]
The test suite is currently failing

**Comment by groony** [comment]
@seanlinsley now only for rails 5 and there is suite that has no relation to this pull request

**Comment by seanlinsley** [comment]
Thanks for that @groony. Would you mind squashing this into a single commit, and also adding a test ensuring that it actually fixes the problem? That test would make sense in [`filters.feature`](https://github.com/activeadmin/activeadmin/blob/master/features/index/filters.feature). This will probably require updating the step definitions to let you set a custom `created_at` time. Let me know if you need help.

File: lib/ransack_ext.rb [hunk 3]
```diff
@@ -9,4 +9,11 @@
   {'equals'=>'eq', 'greater_than'=>'gt', 'less_than'=>'lt'}.each do |old,current|
     config.add_predicate old, arel_predicate: current
   end
+
+  config.add_predicate 'gteq_date',
+    arel_predicate: 'gteq'
```
Comment (timoschilling) [inline lib/ransack_ext.rb]: the `gteq` value should be converted to `<date> 00:00:00`

**Comment by bestori** [comment]
PR seems valid, why does travis fail?

**Comment by constXife** [comment]
So?

**Comment by timoschilling** [comment]
merged by #4504

**Pull request closed without merging**